test/mocks/upstream/mocks.cc

```diff
@@ -33,6 +33,23 @@ MockHostSet::MockHostSet(uint32_t priority) : priority_(priority) {
   }));
 }
 
+std::shared_ptr<const HostSet>
+MockHostSet::filter(std::function<bool(const Host&)> predicate) const {
```

Comment:
having the filter logic duplicated for `HostSetImpl` and `MockHostSet` is a bit annoying. open for suggestions here